src/tests/end-to-end/tests/details-view/cross-origin-iframe.test.ts

```diff
@@ -16,42 +16,73 @@ const testResourceServerConfig: ResourceServerConfig = {
     absolutePath: path.join(__dirname, '../../test-resources/'),
 };
 
-describe('cross-origin iframe and permissions', () => {
+describe('scanning', () => {
     let browser: Browser;
     let targetPage: TargetPage;
     let fastPassAutomatedChecks: DetailsViewPage;
 
-    beforeAll(async () => {
+    beforeAll(() => {
         // we need a second test resource server to get cross-origin content load on the page
         testResourceServer.startServer(testResourceServerConfig);
-
-        browser = await launchBrowser({ suppressFirstTimeDialog: true, addLocalhostToPermissions: true });
-        targetPage = await browser.newTargetPage({ testResourcePath: 'all-cross-origin-iframe.html' });
-        await browser.newPopupPage(targetPage); // Required for the details view to register as having permissions/being open
-
-        fastPassAutomatedChecks = await openAutomatedChecks();
     });
 
     afterAll(async () => {
         testResourceServer.stopServer(testResourceServerConfig);
+    });
 
-        if (browser) {
-            await browser.close();
-            browser = undefined;
-        }
+    describe('with localhost permissions only', () => {
+        beforeEach(async () => {
+            await launchFastPassWithExtraPermissions('localhost');
+        });
+
+        afterEach(async () => {
+            if (browser) {
+                await browser.close();
+                browser = undefined;
+            }
+        });
+
+        it('does not get results from inside cross-origin iframes', async () => {
+            const automatedChecks = await formatPageElementForSnapshot(
+                fastPassAutomatedChecks,
+                fastPassAutomatedChecksSelectors.ruleDetailsGroups,
+            );
+
+            expect(automatedChecks).toMatchSnapshot();
```

Comment:
It's already other tests' responsibility to test the E2E html generated to represent the fastpass cards view, right? Maybe this test should stick to just testing that the expected counts are present, rather than conflating the formatting testing vs the origin permissions testing?

Reply:
(eg, see `it('supports expanding and collapsing rule groups', ...)` in `/src/tests/electron/tests/automated-checks-view.test.ts`)

Reply:
@dbjorge is this closer to what you had in mind?